app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -128,12 +128,45 @@ def process_collection_item(item, key)
       new_result
     end
 
+    def find_host_by_provider_id(provider_id)
+      providers = {
+        "gce" => [ManageIQ::Providers::Google::CloudManager, :name],
+        "aws" => [ManageIQ::Providers::Amazon::CloudManager, :uid_ems],
+      }
+      scheme, instance_uri = provider_id.split("://", 2)
+      prov, name_field = providers[scheme]
+      return nil unless prov
+      instance_name = instance_uri.split('/').last
+
+      instance_details = {name_field => instance_name}
+      prov::Vm.find_by(instance_details)
+    end
+
+    def find_host_by_bios_uuid(new_result)
+      uuid_provider_types = [ManageIQ::Providers::Redhat::InfraManager::Vm.name,
+                             ManageIQ::Providers::Openstack::CloudManager::Vm.name,
+                             ManageIQ::Providers::Vmware::InfraManager::Vm.name]
+      Vm.find_by(:uid_ems => new_result[:identity_system].try(:downcase),
```

Comment:
@enoodle I think we already commented on this. What if `new_result[:identity_system].try(:downcase)` is `nil`?

I think you should handle that case.